importer/importer.go

```diff
@@ -591,18 +591,43 @@ func GetTimestamp(xpath *xpath.Expression, xmlNode xml.Node) *int64 {
 
 func TimestampToSeconds(timestamp string) *int64 {
 	var desiredDateUnix = new(int64)
-	year, _ := strconv.ParseInt(timestamp[0:4], 10, 32)
-	month, _ := strconv.ParseInt(timestamp[4:6], 10, 32)
-	day, _ := strconv.ParseInt(timestamp[6:8], 10, 32)
+	tlen := len(timestamp)
+	if tlen < 8 {
+		return nil
+	}
+	year, err := strconv.ParseInt(timestamp[0:4], 10, 32)
+	if err != nil {
+		return nil
+	}
+	month, err := strconv.ParseInt(timestamp[4:6], 10, 32)
+	if err != nil {
+		return nil
+	}
+	day, err := strconv.ParseInt(timestamp[6:8], 10, 32)
+	if err != nil {
+		return nil
+	}
 	var hour, minute, second int64
-	if len(timestamp) > 8 {
-		hour, _ = strconv.ParseInt(timestamp[8:10], 10, 32)
-		minute, _ = strconv.ParseInt(timestamp[10:12], 10, 32)
-		second, _ = strconv.ParseInt(timestamp[12:14], 10, 32)
-	} else {
-		hour = 0
-		minute = 0
-		second = 0
+	if tlen > 8 {
+		for i := range timestamp {
+			switch i {
+			case 9:
+				hour, err = strconv.ParseInt(timestamp[8:10], 10, 32)
```

Comment:
I wonder if the same issue is true here, as in above, where we want the error checking to be in the `case` statements?